File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        char x = s.charAt(0);
        int cnt1 = 1;
        int cnt2 = 0;
        int answer = 0;
        
        for (int i=1;i<s.length();i++) {
            if (x==' ') {
                x = s.charAt(i);
                continue;
            }
            if (x==s.charAt(i)) {
                cnt1++;
            } else {
                cnt2++;
            }
            if (cnt1==cnt2) {
                answer++;
                x = ' ';
                cnt1 = 1;
                cnt2 = 0;
            }
        }
        if (x!=' ') {
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    first = ""
    cnt1 = 0
    cnt2 = 0
    
    for i in s:
        if first == "":
            first = i
            cnt1 += 1
            continue
        if first == i:
            cnt1 += 1
        else:
            cnt2 += 1
            
        if cnt1 == cnt2:
            answer += 1
            first = ""
            cnt1 = 0
            cnt2 = 0
    if first != "":
        answer += 1
    return answer